app/models/service_provider_identity.rb

```diff
@@ -24,16 +22,16 @@ def deactivate
     update!(session_uuid: nil)
   end
 
-  def sp
-    @sp ||= ServiceProvider.from_issuer(service_provider)
+  def sp_metadata
+    service_provider_record&.metadata || {}
```

Comment:
Curious, if the relationship is a `belongs_to`, could we be guaranteed that `service_provider_record` is non-`nil`?

Reply:
no, because we've seen sometimes SPs over time but these records stay --- `belongs_to` just tells us where we would get the data if it was there, but doesn't provide any strong guarantees or foreign key constraints at the DB level

https://github.com/18F/identity-idp/pull/5187/files#diff-a39056c1e7113debf707209c2f2822d1d7447d55e423535d2130b8b795ced6feR24-R27

^^ that spec is from when we discovered the account page blowing up because an identity in prod linked to an SP that was no longer around